accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,87 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This is rather complicated and inconsistent, which can make for a poor first impression of the .NET platform.
+
+We'd like to try to improve the output path structure.  Desired qualities include:
+
+- Simple and consistent
+- Avoid cases where one output path is nested inside of another one
+- Avoid excessively deep directory hierarchies
+- Get rid of `obj` folder in project root
+
+## Proposed behavior
+
+The new output path format will consist of the following 3 nested folders:
+
+- `artifacts` - All output will go under this folder
+- Output Type - Such as `bin`, `publish`, `intermediates`, or `package`
```

Comment:
How did we settle on `intermediates`? Or maybe not settled but more got to a point that made sense? For many .NET people we have `obj` so might make sense to keep that just to make transition easier. 

Just going to pull from Android Studio here and they also have this sort of pattern with things, but use `intermediates`, `output` and `generated`. However, as @Redth mentions somewhere here, these are long words. So the question for me is can we use another word and save many characters.

Reply:
Also, how did we pick `artifacts`? I see Android Studio uses `build` which is shorter but also more specific? Although, that folder is probably taken by nuget folder names - I know I have a bunch of them...

Reply:
> How did we settle on `intermediates`?

I suggested it because this proposal is already making some significant changes to output paths, and `obj` really doesn't reflect what it contains at all.

In general, I agree that shorter paths are nicer. But then we should be consistent: `package` -> `pkg`, `publish` -> `pub`, `intermediates` -> `int` (?), etc...

Reply:
I agree that they should be short and, in general, shouldn't be acronyms, contractions or abbreviations unless they are well-known accepted. Why not just `temp`? It doesn't get much more intention revealing that than.

Reply:
I've updated the proposal, but am still not sure whether we should go with `obj` or `intermediates`.  I think it probably doesn't really matter, but let me know if you think otherwise :-)